source/StellarObject.cpp

```diff
@@ -155,6 +156,34 @@ double StellarObject::Distance() const
 
 
 
+void StellarObject::UpdateDistanceVisibility(const Ship *ship)
+{
+	if(trueDistanceInvisible < 0.)
+		return;
+
+	for(const auto &clearer : distanceVisibilityClearers)
+		if(ship->Attributes().Get(clearer))
+		{
+			distanceInvisible = -1.;
+			return;
+		}
+
+	double totalMod = 1.;
+	for(const auto &multiplier : distanceVisibilityMultipliers)
+		if(ship->Attributes().Get(multiplier))
+			totalMod *= ship->Attributes().Get(multiplier);
+
+	double totalAdd = 0.;
+	for(const auto &adder : distanceVisibilityAdders)
+		if(ship->Attributes().Get(adder))
+			totalAdd += ship->Attributes().Get(adder);
+
+	distanceInvisible = trueDistanceInvisible * totalMod + totalAdd;
+	distanceVisible = trueDistanceVisible * totalMod + totalAdd;
```

Comment:
~~I cant be bothered to understand all of this adder multiplier and clearers, add comments~~

Reply:
done